pkg/scheduler/event_handler.go

```diff
@@ -130,20 +129,26 @@ func (s *Scheduler) onResourceBindingAdd(obj interface{}) {
 }
 
 func (s *Scheduler) onResourceBindingUpdate(old, cur interface{}) {
-	unstructuredOldObj, err := helper.ToUnstructured(old)
+	t, err := meta.TypeAccessor(old)
 	if err != nil {
-		klog.Errorf("Failed to transform oldObj, error: %v", err)
+		klog.Errorf("Failed to transform oldObj as meta.Type, error: %v", err)
 		return
 	}
 
-	unstructuredNewObj, err := helper.ToUnstructured(cur)
+	oldMeta, err := meta.Accessor(old)
 	if err != nil {
-		klog.Errorf("Failed to transform newObj, error: %v", err)
+		klog.Errorf("Failed to transform oldObj as metav1.Object, error: %v", err)
 		return
 	}
 
-	if unstructuredOldObj.GetGeneration() == unstructuredNewObj.GetGeneration() {
-		klog.V(4).Infof("Ignore update event of object (kind=%s, %s/%s) as specification no change", unstructuredOldObj.GetKind(), unstructuredOldObj.GetNamespace(), unstructuredOldObj.GetName())
```

Comment:
Does this `unstructuredOldObj.GetKind()` always return ""? If yes, is there any bad effect in addition to the log?

Reply:
> Does this unstructuredOldObj.GetKind() always return ""? 

Yes.

Bad effects are minimal because this function only deal with rb and crb, and with or without namespace is enough to explain what is happening here. The reason I needed to change it because I update the `ToUnstructured()` which was obviously affected it.

Reply:
OK. I will leave my question to ToUnstructured() part.